naga/src/compact/mod.rs

```diff
@@ -6,56 +6,76 @@ mod types;
 
 use alloc::vec::Vec;
 
-use crate::arena::HandleSet;
-use crate::{arena, compact::functions::FunctionTracer};
+use crate::{
+    arena::{self, HandleSet},
+    compact::functions::FunctionTracer,
+    ir,
+};
 use handle_set_map::HandleMap;
 
 #[cfg(test)]
 use alloc::{format, string::ToString};
 
-/// Remove unused types, expressions, and constants from `module`.
+/// Remove most unused objects from `module`, which must be valid.
 ///
-/// Assume that the following are used by definition:
+/// Always removes the following unused objects:
+/// - anonymous types, overrides, and constants
+/// - abstract-typed constants
+/// - expressions
```

Comment:
question: Would it be feasible to list the things that _aren't_ compacted, d'y'think?

Reply:
I think that's listed below this? Or else maybe I don't understand what you're looking for.